src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/core/AbstractCraftingRequestResolver.java

```diff
@@ -231,17 +225,17 @@ protected boolean createsCraftingCycle(
             return false;
         }
 
-        return createsCraftingCycle(manager, manager.getRequestForToken(request.getParent()), target, count + 1, reqs);
+        return createsCraftingCycle(manager, manager.getRequestForToken(request.getParent()), target, targetRequest, count + 1);
     }
 
     /**
-     * Check if a building can craft a certain stack.
+     * Check if a building can craft a certain recipe.
      *
      * @param building       the building to check in.
-     * @param stackPredicate predicate used to check if a building knows the recipe.
+     * @param recipe         the recipe to check.
      * @return true if so.
      */
-    public abstract boolean canBuildingCraftStack(@NotNull final AbstractBuilding building, Predicate<ItemStack> stackPredicate);
+    public abstract boolean canBuildingCraftStack(@NotNull final AbstractBuilding building, IRecipeStorage recipe);
```

Comment:
This is more `canBuildingCraftRecipe` now, not `canBuildingCraftStack`.

Reply:
Good point